prefetch.bs

```diff
@@ -67,31 +67,80 @@ spec: RFC8941; urlPrefix: https://www.rfc-editor.org/rfc/rfc8941.html
 
+An <dfn>exchange record</dfn> is a [=struct=] with the following [=struct/items=]:
+* <dfn for="exchange record">request</dfn>, a [=request=]
+* <dfn for="exchange record">response</dfn>, a [=response=] or null
+* <dfn for="exchange record">cookie data</dfn>, a [=list=] of [=tuples=] of [=byte sequences=]
+* <dfn for="exchange record">authentication entry</dfn>, an [=authentication entry=] or null
+
+<div class="note">These records can be used to defer checks that would ordinarily happen during a navigate fetch, and to check for modified [=credentials=].</div>
+
+A <dfn>redirect chain</dfn> is a [=list=] of [=exchange records=].
+
+<div algorithm>
+    To <dfn for="redirect chain">append</dfn> a [=request=] |request| to a [=redirect chain=] |redirectChain|:
+
+    1. Let |cookieData| be the result of [=gathering cookie data=] for |request|'s [=request/URL=] in the cookie store associated with |request| if the user agent is not configured to block cookies for |request|, and an empty [=list=] otherwise.
+    1. Let |authenticationEntry| be null.
+    1. If there's an [=authentication entry=] for |request| and |request|'s [=request/current URL=] does not [=include credentials=], then set |authenticationEntry| to that [=authentication entry=].
+    1. [=list/Append=] a new [=exchange record=] whose [=exchange record/request=] is a copy of |request|, [=exchange record/response=] is null, [=exchange record/cookie data=] is |cookieData|, and [=exchange record/authentication entry=] is |authenticationEntry| to |redirectChain|.
+</div>
+
+<div algorithm>
+    To <dfn for="redirect chain">update the response</dfn> for a [=redirect chain=] |redirectChain| given a [=request=] |request| and [=response=] |response|:
+
+    1. [=Assert=]: |redirectChain| is not [=list/empty=].
+    1. [=Assert=]: |redirectChain|'s last element's [=exchange record/request=] is |request| and its [=exchange record/response=] is null.
+    1. Set |redirectChain|'s last element's [=exchange record/response=] to |response|.
+</div>
+
+<div algorithm>
+    A [=redirect chain=] |redirectChain| <dfn for="redirect chain">has updated credentials</dfn> if the following steps return true:
+
+    1. [=list/For each=] |exchangeRecord| of |redirectChain|:
+        1. Let |request| be |exchangeRecord|'s [=exchange record/request=].
+        1. Let |cookieData| be the result of [=gathering cookie data=] for |request|'s [=request/URL=] in the cookie store associated with |request| if the user agent is not configured to block cookies for |request|, and an empty [=list=] otherwise.
+        1. If |cookieData| is not identical to |exchangeRecord|'s [=exchange record/cookie data=], then user agents must return true. User agents may also return true if a modification to the cookies applicable to |request| has occurred in some other way.
```

Comment:
Maye "in some other way despite the cookie data being identical"?

Reply:
added words to that effect